src/types/transaction.cairo

```diff
@@ -137,12 +140,29 @@ impl ByteArraySnapSerde of Serde<@ByteArray> {
     }
 }
 
+
+impl ByteArraySnapHash<S, +HashStateTrait<S>, +Drop<S>> of Hash<@ByteArray, S> {
+    #[inline]
+    fn update_state(mut state: S, value: @ByteArray) -> S {
+        let mut index = 0;
```

Comment:
You can use `self.serialize` to convert into array of felt252 and then add felts one by one to the state.